Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -49,45 +51,74 @@ namespace AzToolsFramework
             m_transform = nullptr;
         }
 
-        void EditorComponentBase::OnPrepareForAdditionToEntity(AZ::Entity* entity)
+        void EditorComponentBase::OnAfterEntitySet()
         {
-            if (entity && m_alias.empty())
+            AZ_Assert(m_entity, "OnAfterEntitySet - Entity should not be nullptr.");
+
+            // Sets up only if the component alias is empty to avoid setting the alias twice.
+            // For a component added as a pending component, there are two places trying to call this function to set up the alias:
+            // - EditorEntityActionComponent::AddExistingComponentsToEntityById()
+            // - Entity::AddComponent()
+            if (m_alias.empty())
             {
-                AZStd::unordered_set<AZStd::string> serializedIdentifiersMatchingType;
+                AZStd::string newIdentifier = GenerateComponentSerializedIdentifier();
+                SetSerializedIdentifier(AZStd::move(newIdentifier));
+            }
+        }
 
-                for (const AZ::Component* componentInEntity : entity->GetComponents())
-                {
-                    if (componentInEntity == this)
-                    {
-                        // If the alias is not empty and the component is already added to the entity, there's nothing to do here.
-                        return;
-                    }
-                    if (RTTI_GetType() == componentInEntity->RTTI_GetType())
-                    {
-                        serializedIdentifiersMatchingType.emplace(componentInEntity->GetSerializedIdentifier());
-                    }
-                }
+        AZStd::string EditorComponentBase::GenerateComponentSerializedIdentifier()
+        {
+            AZ_Assert(m_entity, "GenerateComponentSerializedIdentifier - Entity should not be nullptr.");
```

Comment:
if m_entity is null here, this will cause a null pointer reference, and crash later.   Asserts keep going - they do not stop execution or return.

Perhaps add a bail out clause (return early if no entity?), or perhaps move this assert later down in the function to right before it uses m_entity, and maybe add a way for it to safely proceed, ie, by not calling any entity-related functions, skipping that part, so that in the worst case, the entity-related stuff doesn't get involved in making the identifer.